src/connection.js

```diff
@@ -859,7 +918,18 @@ class Connection extends EventEmitter {
     this.debug.payload(function() {
       return preloginPayload.toString('  ');
     });
-
+    if (0 != preloginPayload.fedAuthRequired && 1 != preloginPayload.fedAuthRequired) {
+      this.emit('connect', ConnectionError('Server sent an unexpected value for FedAuthRequired PreLogin Option. Value was' + preloginPayload.fedAuthRequired, 'EFEDAUTH'));
+      return this.close();
```

Comment:
This error would be triggered for all older versions of SQL Server that doesn't support fedAuth 🤔 How about changing the condition such that error is triggered only when client requests fedAuth and server responds with something other than 1 or 0?